### DISCUSSION TITLE
useEffect returns auto cancelled

### DISCUSSION THREAD

**Discussion by Marius555** [pocketbase/pocketbase#3491] [Q&A]
im using next 13 useEffetct and im trying to call
```js
await pb.collection('users').listAuthMethods()
```
but im getting auto cancelled...i know that is because of useEffect running on every re-render, i know i can disable  auto cancel, but that defeats the whole purpose of it... so i want to ask what is the correct way to call
```js
await pb.collection('users').listAuthMethods()
```
because useEffect re-renders component 4 times and i cant  resolve a promise of the first request, because its was canceled and only last was resolved...im a bit new in javascript and react so i really struggling to resolve this...

**Comment by ganigeorgiev** [comment]
> useEffect re-renders component 4 times

Is a single instance of the component being rerendered or there are multiple instances of the same component?

If it is the first case, then auto cancellation works as intended and you can safely ignore the aborted request errors. To distinguish between auto cancelled error and an API error, you can check the `err.isAbort` field:

```js
// async/await syntax
try {
    await pb.collection("users").listAuthMethods()
} catch (err) {
    if (!err.isAbort) {
        console.warn("non cancellation error:", err)
    }
}

// or the same with the regular Promise chain syntax
pb.collection("users").listAuthMethods()
    .catch((err) => {
        if (!err.isAbort) {
            console.warn("non cancellation error:", err)
        }
    })
```

If it is the second case, aka. you want the request to be executed multiple times without cancelling previous pending calls, then you'll need either to disable the auto cancellation for the specific request by setting `requestKey: null` OR assign a unique `requestKey` string identifier for each different call.

```js
pb.collection("users").listAuthMethods({
    // disable auto cancellation only for this particular request
    requestKey: null,

    // or assign a unique requestKey for this call
    requestKey: "something...",
})
```
